finalfrontier/src/deps.rs

```diff
@@ -116,6 +118,32 @@ impl<'a> DependencyIterator<'a> {
         }
     }
 
+    /// Construct a `DependencyIterator` and apply parameters given in `config`.
+    pub fn new_from_config(
+        graph: &'a DepGraph<'a>,
+        config: DepembedsConfig,
+    ) -> Box<Iterator<Item = (usize, Dependency)> + 'a> {
+        let iter = DependencyIterator {
```

Comment:
Just a teensie comment: since this is exactly the same as `new`, just call new here to construct the initial iterator.

Other than that, squash and merge :).

Reply:
Done!